src/main/java/sirius/db/jdbc/BaseSQLQuery.java

```diff
@@ -133,8 +150,10 @@ public Optional<Row> first() throws SQLException {
     @Nullable
     public Row queryFirst() throws SQLException {
         ValueHolder<Row> result = ValueHolder.of(null);
-        iterateAll(result, Limit.singleItem());
-
+        doIterate(row -> {
+            result.accept(row);
+            return true;
```

Comment:
we could actually also return false here as there shouldn't be anything left in the result set anyway...